### DISCUSSION TITLE
Zugriff auf TIA, was kann die Lib

### DISCUSSION THREAD

**Discussion by Nick135** [dotnetprojects/DotNetSiemensPLCToolBoxLibrary#173] [Q&A] [answered]
Was ist mit den in der Lib enthaltenen Funktionen im Bezug auf TIA möglich?
Ist eine entsprechende TIA Installation notwendig? Für TIA Openness wird diese ja benötigt.

**Comment by jogibear9988** [comment, marked as answer]
Die Bibliothek verwendet die Openness DLLs von TIA, d.h. TIA muss installiert sein.

Man kann im Endeffekt die selbe API verwendet mit der man auf Step7 Bausteine Zugreift um auf ein TIA Projekt zuzugreifen.
Aber wir unterstützen natürlich nur Teile der Openness API. (Das was ich halt gebraucht habe)

Ziel am Anfang war mal das TIA File Format Nativ zu unterstützen, aber das ist ein riesen Aufwand, das erst mal zu Analysieren, dann ändert sich in jeder Version alles mögliche. Ich bin damals damit nicht weitergekommen, und habe dafür auch z.Zt. keine Zeit.
Der Code der dafür schon existiert ist hier: 
https://github.com/dotnetprojects/DotNetSiemensPLCToolBoxLibrary/tree/master/LibNoDaveConnectionLibrary/Projectfiles/TIA
(ist aber nicht viel)
